test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -409,20 +409,24 @@ public void ParseFilterWithNullEnumValue()
             convertNode.Source.ShouldBeConstantQueryNode((object)null);
         }
 
-        [Fact]
-        public void ParseFilterCastMethod1()
+        [Theory]
+        [InlineData("cast(NS.Color'Green', 'Edm.String') eq 'blue'")]
+        [InlineData("cast(NS.Color'Green', Edm.String) eq 'blue'")]
+        public void ParseFilterCastMethod1(string filterQuery)

```

Comment:
The test name contains XXX1 AND XXX2 doesn't make sense. Can you rename them?

And can combine 'ParseFilterCastMethod1' and 'ParseFilterCastMethod2'?